weavetest/assert/assert.go

```diff
@@ -0,0 +1,36 @@
+package assert
+
+import (
+	"reflect"
+	"testing"
+)
+
+// NoErr fails the test if given error is not nil.
+func NoErr(t testing.TB, err error) {
+	t.Helper()
+	if !errIsNil(err) {
+		t.Fatalf("want the error to be nil, got %+v", err)
```

Comment:
These are nice helpers, I have a comment about the language. Following the testify library conventions, which seemed well though out:

`assert.Xyz()` will call `t.Errorf()` if the assertion fails.
`require.Xyz()` will call `t.Fatalf()` if the assertion fails.

My actual two main use cases were:
 `require.NoErr()` -> to abort the test if some setup code errors and
 `assert.Equal()` -> to check if is what I expected.

The `NoErr()` function does what I expect, but named differently. Maybe make a second package as @alpe suggested.

Reply:
I dont know how to feel about making a package with one function just for the namespace. I guess nothing harmful in that. It feels like it is a bit forced.

I think the notation of `assert` marking the test as failed and `require` failing the test instantly is testify's invention. In other languages I kow, `assert` is always throwing an error which in result is finishing the test.
I cannot think of a case where I had to use `Errorf` and `Fatalf` was not good enough.

Should I create a `require` package as well and repeat the code there?

Reply:
I didn't want to create any new package.. this was @alpe 's idea.
But i see how `require.NoError(t, err)` is easier to write then `weavetest.requireNoError(t, err)`.

But I will accept making assert call `Fatalf` with your arguments for it. Also makes this simpler. In such case, code is good as is.

Reply:
ja, `require` and `assert` are testify semantics. I am 🙆‍♂️ with this `assert` implementation to fail fast and exit test execution.

Reply:
note: `NoErr` is a specialization of `is nil` for errors only. I wouldn't mind `assert.IsNil(t, err)` instead.

Reply:
Should I change to `IsNil`?

Reply:
I guess the expectation is `NoError` as it was mentioned in #249 . If anybody wants `IsNil` instead it can be added easily. Just noted the fact but did not want to create confusion.